src/main/java/com/github/kiulian/downloader/downloader/request/RequestSearchResult.java

```diff
@@ -0,0 +1,100 @@
+package com.github.kiulian.downloader.downloader.request;
+
+import java.util.*;
+
+import com.github.kiulian.downloader.base64.Base64Encoder;
+import com.github.kiulian.downloader.model.search.SearchResult;
+import com.github.kiulian.downloader.model.search.field.*;
+
+public class RequestSearchResult extends Request<RequestSearchResult, SearchResult> {
+
+    private final String query;
+    private Map<Integer, SearchField> filterFields = new HashMap<>();
+    private SortField sortField;
+
+    public RequestSearchResult(String query) {
+        super();
+        this.query = query;
+    }
+
+    public String encodeParameters() {
+        int filterLength = 0;
+        List<SearchField> filters = null;
+        if (!filterFields.isEmpty()) {
+            filters = new ArrayList<>(filterFields.values());
+            filters.sort((f1, f2) -> f1.category() - f2.category());
+            for (SearchField filter : filters) {
+                filterLength += filter.length();
+            }
+        }
+        
+        if (sortField == null && filterLength == 0) {
+            return null;
+        }
+        
+        int length = filterLength;
+        if (sortField != null) {
+            length += 2;
+        }
+        if (filters != null) {
+            length += 2;
+        }
+        
+        final byte[] bytes = new byte[length];
+        int i = 0;
+        if (sortField != null) {
+            bytes[i++] = 8;
+            bytes[i++] = sortField.value();
+        }
+        if (filters != null) {
+            bytes[i++] = 18;
+            bytes[i++] = (byte) filterLength;
+            for (SearchField filter : filters) {
+                System.arraycopy(filter.data(), 0, bytes, i, filter.length());
+                i += filter.length();
+            }
+        }
+        
+        String encoded = Base64Encoder.getInstance().encodeToString(bytes);
+        return encoded.replace("=", "%253D");
+    }
+
+    public String query() {
+        return query;
+    }
+
+    public RequestSearchResult select(SearchField... field) {
```

Comment:
`filter()` sounds better, because basically this method adds filter fields, isnt it?

Reply:
Yes it does, it was the only method at first.
Then I added methods for each type of field, and `filter` was for `TypeField`, then it became `match` (which is awful by the way, if you have ideas...).
So I agree, `filter()` is better.
I think I'll move the first 6 `FeatureField` members to a `FormatField` class, and add a `format()` or `filterFormat()` method.

Reply:
Previously I told that I prefer separate method for each search param, but after review I am ok with single `filter` method, more intuitive, because it is similar to what user does in web youtube. 
Its up to you, you can remove other methods and keep only `filter`

Reply:
`filter` and other filter methods are all quite useful and intuitive.
I prefer keeping all of them.